result.go

```diff
@@ -175,3 +202,14 @@ func newResultObject(t reflect.Type) (resultObject, error) {
 }
 
 func (ro resultObject) Produces() map[key]struct{} { return ro.produces }
+
+func (ro resultObject) Extract(c *Container, v reflect.Value) error {
+	for _, f := range ro.Fields {
+		if err := f.Result.Extract(c, v.Field(f.Index)); err != nil {
+			// In reality, this will never fail because none of the fields of
+			// a resultObject can be resultError.
```

Comment:
To be clear, this is not possible to reproduce in a test because it requires creating an invalid resultObject. Yes, I could hand-build an invalid resultObject but that sort-of defeats the purpose.

I'm not sure if we have agreement around how to handle testing situations like this.

Reply:
This doesn't seem any different from the unreachable cases in the type switches we've introduced; since we've already introduced a precedent for it, we can panic if this code is really unreachable.